### DISCUSSION TITLE
Extend Now Macro

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6872] [Ideas]
Does it make sense to extend now macro a little bit for example to return date 14 days from now, or dates 50 days before today

```
<<now "YYYY-0MM-0DD"  "-14">>

<<now "YYYY-0MM-0DD"  "50">>
```

Some refs:
-https://github.com/Jermolene/TiddlyWiki5/issues/6267

**Comment by pmario** [comment]
If you open the tiddler tomorrow, you will get a different date. ... Is that your intention?

**Reply by kookma** [reply]
My purpose is to set due date like below:

```
<$let due=<<now "YYYY.0MM.0DD" "20">> >
```

Or find the weekday 15 days ago!

**Reply by pmario** [reply]
The problem with `<<now` for this purpose is, that it will always use the current date when it is created. So it will show "now + 14days" but now is always adjusted to the current date.

So what you probably want is a toolbar button, that lets you insert a "delta" and creates a "hardcoded" date string.

**Reply by kookma** [reply]
> So it will show "now + 14days" but now is always adjusted to the current date.

That is through, but it used with a button and the produced date is inserted as literal string in due-date field!

By the way, it means to be able to do date/time arithmetic in TW.
